File: SunwooChoi/2022-11-09-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        c_dict, t_dict = {}, {}
        for c in s:
            c_dict[c] = 1 + c_dict[c] if c in c_dict else 1
        
        for c in t:
            t_dict[c] = 1 + t_dict[c] if c in t_dict else 1

        return t_dict == c_dict

File: DonRyu/2022-11-09-Valid Anagram.js
var isAnagram = function (s, t) {
	if (s.length != t.length) return false;
	const hashTable = {};

	for (let i = 0; i < s.length; i++) {
		if (!hashTable[s[i]]) {
			hashTable[s[i]] = 0;
		}
		hashTable[s[i]]++;
	}

	for (let j = 0; j < t.length; j++) {
		if (!hashTable[t[j]]) {
			return false;
		}
		hashTable[s[j]]--;
	}

	return true;
};
